PP-426/review-documentation (#1)
franciscotobar/devportal: guides/rif-relay/integrate.md

In any case, you'll need to have the server installed and running. To achieve this please refer to the following guides:
1. [RIF Relay Installation](/guides/rif-relay/install/)
2. [RIF Relay Launch](/guides/rif-relay/deployment/)

### RIF Relay SDK

> To simplify the integration process a SDK was developed, the SDK is a library that needs to be used from a dApp that will provide us a way to sign transactions. The SDK job is to provide a simple access to the general functions from the RIF Relay system. Below are the steps to create an instance of the sdk:

Reviewer: ```suggestion
To simplify the integration process a SDK was developed,
the SDK is a library that needs to be used from a dApp that will provide us a way to sign transactions.
The SDK's job is to provide a simple access to the general functions from the RIF Relay system.
To create an instance of the SDK, follow the steps below:
```

Author: updated

    };

    const contractAddresses: RelayingServicesAddresses = {
        relayHub: '0xE0825f57Dd05Ef62FF731c27222A86E104CC4Cad',
        smartWallet: '0x73ec81da0C72DD112e06c09A6ec03B5544d26F05',
        smartWalletFactory: '0x03F23ae1917722d5A27a2Ea0Bcc98725a2a2a49a',
        smartWalletDeployVerifier: '0x1eD614cd3443EFd9c70F04b6d777aed947A4b0c4',
        smartWalletRelayVerifier: '0x5159345aaB821172e795d56274D0f5FDFdC6aBD9',
        // TODO: Why aren't these addresses required? we may set them as optional
        penalizer: '',
        customSmartWallet: '',
        customSmartWalletFactory: '',
        customSmartWalletDeployVerifier: '',
        customSmartWalletRelayVerifier: '',
        sampleRecipient: ''
    };

    const web3 = new Web3(<RSK_NODE_ENDPOINT>);

    const relayingServices = new DefaultRelayingServices(web3);
    await relayingServices.initialize(config, contractAddresses, {
        loglevel: 1
    });
```
> All the contracts addresses and the chainId are from RSK Node Regtest

Where variables are:

  * **RSK_CHAIN_ID**: the RSK node chain id where is running (ex: 33).
  * **RIF_RELAY_SERVER_URL**: the RIF Relay Server url (ex: http://localhost:8090).

>   * **RSK_NODE_ENDPOINT**: the RSK node endpoint where is running (ex: http://localhost:4444).

Reviewer: ```suggestion
  * **RSK_CHAIN_ID**: The RSK node chain ID which is running (e.g. `33`).
  * **RIF_RELAY_SERVER_URL**: The RIF Relay Server URL (e.g. `http://localhost:8090`).
  * **RSK_NODE_ENDPOINT**: The RSK node endpoint which is running (e.g. `http://localhost:4444`).
```

Author: updated

    // Step 3
    const relayTrxOpts: RelayingTransactionOptions = {
        smartWallet,
        unsignedTx,
        tokenAddress: <TOKEN_ADDRESS>,
        tokenAmount: <AMOUNT_OF_TOKENS>
    };

    // Step 4
    const relayintResult: RelayingResult = await relayingServices.relayTransaction(
        relayTrxOpts
    );
```

Where variables are:

  * **SMART_WALLET**: the generated smart wallet from the deployment process.
  * **TOKEN_ADDRESS**: the token contract address.
  * **AMOUNT_OF_TOKENS**: string containing the amount of tokens in decimal unit.

### Custom worker replenish function

Each relayed transaction is signed by a Relay Worker account. The worker accounts are controlled by the Relay Manager. When a relay worker signs and relays a transaction, the cost for that transaction is paid using the funds in that worker's account. If the transaction is not subsidized, then the worker is compensated with tokens.

Worker accounts must always have some minimum balance to pay gas for the transaction. These balances can be managed by implementing a replenishment strategy. The Relay Manager can use the strategy to top off a relay worker's account when the balance gets too low.

> We provide a default implementation for a replenishment strategy. RIF Relay solution integrators can implement their own replenish strategy.

Reviewer: ```suggestion
We provide a default implementation for a replenish strategy.
RIF Relay solution integrators can implement their own replenish strategy.
```

Author: updated

To implement and use your own replenish strategy:

1. In the folder `src` from the RIF Relay Server project, open `ReplenishFunction.ts` with a text editor.
2. On the function `replenishStrategy` write your new replenish strategy.
3. Re build the project `npm run build`

> 4. Add the command `--customReplenish` when running a RIF Relay Server or change the config json file to set `customReplenish` on true.

Reviewer: ```suggestion
4. Add the command `--customReplenish` when running a RIF Relay Server or change the config JSON file to set `customReplenish` on true.
```

Author: updated
